common/import_.py

```
@@ -182,26 +191,34 @@ def add_created_activities(dbname, activities_to_create):
     bw2data.Database(dbname).register()
 
     for activity_data in activities_data:
-        if "add" in activity_data:
-            add_average_activity(activity_data, dbname)
-        if "replace" in activity_data:
-            add_variant_activity(activity_data, dbname)
-
-
-def add_average_activity(activity_data, dbname):
-    """Add to the database dbname a new activity : the weighted average of multiple activities
-
-    Example : the average activity milk "Cow milk, organic, system n°1, at farm gate/FR U" is the
-    weighted average of the activities 'Cow milk, organic, system n°1, at farm gate/FR U' from
-    system 1 to 5
+        logger.info(
+            f"-> Creating activity {activity_data.get('activityCreationType')} '{activity_data.get('alias')}'"
+        )
+        if activity_data.get("activityCreationType") == "fromScratch":
```

Using a string comparaison here is dangerous. The day you decide to change your labels, for example "fromScratch" to "from-scratch" you will have not forget to do it everywhere or it will fail silently.

Here you should use a python Enum and more specifically an StrEnum https://docs.python.org/3/library/enum.html#enum.StrEnum

```python
from enum import StrEnum

class ActivityFrom(StrEnum):
    EXISTING = "fromExisting"
    SCRATCH = "fromScratch"
```

And your comparison would become:
```python
if activity_data.get("activityCreationType") == ActivityFrom.SCRATCH:
```

It will allow you to rename your label values as you want and will prevent you from misspelling the "fromScratch" string: if you use an enum value that doesn’t exist the compiler will complain.

And on a side not, it’s unusual to have values with camelCase spelling. Using "from_scratch" or "from-scratch" would be more common.

---

good point, done ✅